schema.xsd

```diff
@@ -166,20 +168,52 @@
         <xs:attribute name="relevance" type="xs:decimal" use="required"/>
     </xs:complexType>
 
+    <!-- Range slider filters. Filters "min"/"max" may not have any values if there is nothing to filter. -->
     <xs:complexType name="range">
         <xs:sequence>
-            <xs:element name="min" type="xs:decimal"/>
-            <xs:element name="max" type="xs:decimal"/>
+            <xs:element name="min">
+                <xs:simpleType>
+                    <xs:union>
+                        <xs:simpleType>
+                            <xs:restriction base='xs:string'>
+                                <xs:length value='0'/>
+                            </xs:restriction>
+                        </xs:simpleType>
+                        <xs:simpleType>
+                            <xs:restriction base='xs:decimal'>
+                            </xs:restriction>
+                        </xs:simpleType>
+                    </xs:union>
+                </xs:simpleType>
+            </xs:element>
+            <xs:element name="max">
+                <xs:simpleType>
+                    <xs:union>
+                        <xs:simpleType>
+                            <xs:restriction base='xs:string'>
+                                <xs:length value='0'/>
+                            </xs:restriction>
+                        </xs:simpleType>
+                        <xs:simpleType>
+                            <xs:restriction base='xs:decimal'>
+                            </xs:restriction>
+                        </xs:simpleType>
+                    </xs:union>
+                </xs:simpleType>
+            </xs:element>
         </xs:sequence>
     </xs:complexType>
 
     <xs:complexType name="filter">
         <xs:sequence>
+            <!-- Optional itemCount until a "show more filters" link should appear, only relevant for type "select". -->
+            <xs:element name="itemCount" type="xs:int" minOccurs="0" maxOccurs="1" />
+
             <!-- Optional CSS class from filter config. -->
-            <xs:element name="cssClass" type="xs:string" minOccurs="0"/>
+            <xs:element name="cssClass" type="xs:string" minOccurs="0" maxOccurs="1" />
```

Comment:
`1` is the default value of `maxOccurs`, so it shouldn't be necessary to specify it explicitly.